commons/src/main/java/org/infinispan/commons/dataconversion/Encoder.java

```diff
@@ -0,0 +1,30 @@
+package org.infinispan.commons.dataconversion;
+
+/**
+ * Used to convert data between read/write format and storage format.
+ * @since 9.1
+ */
+public interface Encoder {
```

Comment:
It could make sense @anistor. I had them separated initially but put together for convenience reasons. The main advantage of having them separated would be to avoid creating an ```Encoder``` for every combination of key/value, like UTF8KeyOnlyEncoder, ProtostreamKeyUTF8ValueEncoder  :smile:  We don't have much of this now, but this can become an issue